packages/diagrams/lib/src/future_builder.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:diagram_capture/diagram_capture.dart';
+import 'package:flutter/material.dart';
+
+import 'diagram_step.dart';
+
+const String _futureBuilder = 'future_builder';
+const String _futureBuilderError = 'future_builder_error';
+const Duration _futureDuration = Duration(seconds: 1);
+const Duration _pauseDuration = Duration(seconds: 1);
+final Duration _totalDuration = _futureDuration + _pauseDuration;
+
+class FutureBuilderDiagram extends StatefulWidget implements DiagramMetadata {
+  const FutureBuilderDiagram(this.name);
+
+  @override
+  final String name;
+
+  @override
+  _FutureBuilderDiagramState createState() => _FutureBuilderDiagramState();
+}
+
+class _FutureBuilderDiagramState extends State<FutureBuilderDiagram> {
+  Future<String> _calculation;
+
+  @override
+  void initState() {
+    if (widget.name == _futureBuilder) {
+      _calculation = Future<String>.delayed(
+        _futureDuration,
+        () => 'Data Loaded',
```

Comment:
We could add color & icons here too to emphasize the result, like green text and a check mark for success, and red with an X for failure. Feel free to have fun with it. :)

Reply:
Another example: smiling/frowning faces. 😝

Reply:
Haha, sounds good! I'm always on the fence of how fancy to get with these things to keep the examples short and to the point, but yah -- let's have some fun 😸 🎉